### DISCUSSION TITLE
Ally: Chatbot for interview scheduling

### DISCUSSION THREAD

**Discussion by garimasingh128** [scaleracademy/open-source-projects#36] [Ideas]
# ALLY:
Ally is a conjunction that encapsulates effective features that will ease the duties of interviewers and hiring managers significantly by reducing the workload and create a positive customer base for the brand. 

# ISSUES IN THE HIRING PROCESS:
- Finding common availability is challenging and time-consuming. This includes getting calendar availability of hiring managers, booking rooms in case of in-person interviews, and ensuring invites have the right links for virtual interviews. 
- Attracting the right candidates for the job who has the suitable skills that fit the Job description is a tedious process still done manually by HRs’. 
- Companies can use recruitment data and metrics to constantly improve their recruiting process and make more informed decisions. But collecting and processing data can be a hassle. 
- Ensuring a good candidate interview experience is of utmost importance, all of which solely depends on the HR Team. A positive candidate experience can enhance the employer brand and encourage good candidates to apply and accept job offers.
- Fair recruiting process: Sometimes certain good candidates more suitable for the opening get left out solely due to human errors in the recruiting process such as the HR forgot to reach out to them on time or the calendar invite for the interview expired before time etc.
- Hiring teams need to communicate fast, evaluate candidates easily and know what’s going on every step of the way. Recruiters are tasked with coordinating all this communication.

# THE IDEA- ALLY 
- Connecting right candidates to the recruiters: Recruiters need to be able to ﬁnd these candidates and connect with them to discuss the position further. Ally is designed to source these hidden talents by using AI and NLP (natural language processing) to scan online resumes and proﬁles to send valuable leads to recruiters. 
- To help streamline the initial interview process and narrow down the applicant pool even further. Ally can be used for initial short interviews via chatbot texts. 
- An AI-powered chatbot that asks various questions and analyzes the candidate’s answers through NLP: Provides ﬂexibility for the candidate to answer the questions on their own time. It can also save recruiters precious time in the qualiﬁcation process and narrow down the number of candidates who are invited for an in-person interview.

# INNOVATION IN OUR PROPOSED SOLUTION
- Chatbot provides faster and clearer replies rather than HR emails. This builds a strong employer brand that makes great candidates chase your company for good. 
- Innovation in face to face interviews: AI lacks human emotions like empathy and are mostly unbiased, which means that it can help recruiters make a decision that is based in facts and data. We can extend Ally to assess the candidate’s facial expressions, body language, sentiment and knowledge for a smarter hiring decision. 
- The chatbot can be introduced as a plugin to the already existing career portal.  Candidates who move to the further rounds will be provided with a login ID and password which will allow them access to the Ally Tool, and they can have 2 way conversations with the AI-figured Hiring manager. A study by Bersin and Deloitte found that the single biggest factor contributing to recruiter performance is their relationships with their hiring managers. And the key to building stronger relationships is having more productive conversations. 
- In case of any specific needs, an email or contact information of a human figure will be provided too.

# WORKFLOW OF THE APP
1. Input: Excel sheet from the recruiter, availability timings of the interviewers and room reservation Excel sheet in case of offline interviews. 
2. Output: A direct link to the online meeting sent to the email of the candidate in case he agrees on the possible time slot. Else, another possible slot, room or involvement of someone else is sought.

# IMPLEMENTATION 
## Scope of the solution: 
Mobile devices and laptops are increasingly being used as platforms for interactive services.  The demand for a single integrated, easy-to-use, access to information applications and people within a network of users is increasing more and more. Therefore, 'Ally' will be a perfect partner for eradication of present problems of candidates and recruiters alike. Our mobile portal is designed to increase accessibility to almost everything that a candidate needs. Once logged in, they will have access to a variety of tools and information. AI can make the job search and talent acquisition process easier and more enjoyable for both parties for multiple reasons.
- The application currently has an Android based interface only. 
- When the user first authenticates himself, he lands on a dashboard which has several options based on his/her interest. 
- The working of the app is fuss-free and smooth in the sense that it uses completely hassle-free methods of storage and authentication. 
- The UI is currently under development, but we intend to follow a same color scheme and pattern throughout the app.

# TECHNOLOGY STACK
1. OPERATING SYSTEM: ANDROID 4.0+ iOS 9.0+ Web 
2. BACKEND DATABASE: GOOGLE FIREBASE (for lower scale)
3. PROGRAMMING LANGUAGE: JAVA 8 or JavaScript SQL XML 
4. DESIGN SOFTWARES: ADOBE XD FIGMA
 
```
In a nutshell: 
Recruitment as a career is the epitome of the phrase “trying to ﬁnd a needle in a haystack”. RECRUITERS CALL FOR AN 'ALLY' AND WE HOPE THAT OUR APPLICATION COMPLETES THE CIRCLE.
Thank you!
```

**Comment by shivaamm** [comment]
hey @garimasingh128 the idea is really great but how the data will be collected by the chatbot and on what basis it will filter. Acc to the idea using chatbot candidate will be answering a few questions but the questions answered will be in an spoken language or even if it's an code how the code will be analysed by the chatbots and what stack will be used.........a lot of questions I have.
I liked the idea and it will be very useful but Implementation seems difficult.
Would love to know more about it :)
